src/rooms/RestartRoom.ts

```diff
@@ -55,6 +60,12 @@ export class RestartRoom extends Room<RestartRoomState> {
   onLeave(client: Client, consented: boolean) {
     console.log(client.sessionId, "left room", this.roomId);
     this.state.players.delete(client.sessionId);
+
+    this.clock.setInterval(() => {
```

Comment:
This will also fail. Assume that 1 player left at T1, then this timer gets created. That player or some other player rejoins at T5 but leaves at T8. Ideally, the room should get disposed at T18, but in your case it will dispose at T11. 
To correct: this.clock.setInterval should be inside the if condition, not outside, and this timer also should get deleted in case some player joins.